File: math1/math1_1212_8_to_2.py
#if 0
# Ive attended the code.plus class (https://code.plus/)
# And I've referred it's codes
# And core parts can be come fully from the class.
#endif

#!/usr/bin/python3
eight = ["000","001","010","011","100","101","110","111"]
s = input()
start = True
ans = ''
if s == '0':
    ans = '0'
for ch in s:
    n = ord(ch)-ord('0')
    if start and n < 4:
        if n == 0:
            continue
        elif n == 1:
            ans += '1'
        elif n == 2:
            ans += '10'
        elif n == 3:
            ans += '11'
        start = False
    else:
        ans += eight[n]
        start = False
print(ans)

File: math1/math1_1212_8_to_2.cpp
#if 0
# Ive attended the code.plus class (https://code.plus/)
# And I've referred it's codes
# And core parts can be come fully from the class.
#endif

#include <iostream>
#include <string>
#include <cstdio>
using namespace std;
string eight[8] = 
{"000","001","010","011","100","101","110","111"};
int main() {
    string s;
    cin >> s;
    bool start = true;
    if (s.length() == 1 && s[0]-'0'==0) { // len:1
        cout << "0";
    }
    for (int i=0; i<s.length(); i++) {
        int n = s[i]-'0';
        if (start == true && n < 4) { // len:2
            if (n == 0) {
                continue;
            } else if (n == 1) {
                cout << "1";
            } else if (n == 2) {
                cout << "10";
            } else if (n == 3) {
                cout << "11";
            }
            start = false;
        } else { // len:3
            cout << eight[n];
            start = false;
        }
    }
    return 0;
}